MailContainerTest/Data/IMailContainerDataStore.cs

```diff
@@ -0,0 +1,10 @@
+﻿using MailContainerTest.Types;
+
+namespace MailContainerTest.Data
+{
+    public interface IMailContainerDataStore
```

Comment:
I would improve this interface by making the methods return `Task<MailContainer>` and `Task`, respectively, to allow for async implementations in future.